### DISCUSSION TITLE
Can we split logger creation and logger display?

### DISCUSSION THREAD

**Discussion by nickosh** [hoffstadt/DearPyGui_Ext#5] [Support]
Hi! Thanks for great UI framework! I have a little confusing what logger object creation also works like logger window displayer:

- Preffer to inject logger into main window. So I must create logger into UI module and only then it can be transfered to another modules;
- Can't write logs before UI creation;
- Very hard to properly split code on modules;

This architecture was built on purpose? Feeling like old design was more flexible.

**Comment by hoffstadt** [comment]
Absolutely!
